File: iact-check-core/src/main/kotlin/ch/iact/iactcheck/dto/QuestionCategoryDTO.kt
package ch.iact.iactcheck.dto

data class QuestionCategoryDTO(
    val id: Long,
    val checkId: Long,
    val title: String,
    val language: LanguageDTO,
    val numberOfPossibleOutcomesToShow: Int,
    val rangeQuestions: List<RangeQuestionDTO>,
    val imageQuestions: List<ImageQuestionDTO>,
    val possibleOutcomes: List<PossibleOutcomeDTO>,
    val possibleOutcomesDisplayType: String
)

File: iact-check-webapp/src/main/resources/webapp/src/app/shared/dtos/question-category-dto.ts
import { PossibleOutcomeDTO } from './possible-outcome-dto';
import { RangeQuestionDTO } from './range-question-dto';
import { LanguageDTO } from './language-dto';
import { ImageQuestionDTO } from './image-question-dto';
import { PossibleOutcomesDisplayType } from './possible-outcomes-display-type';

export interface QuestionCategoryDTO {
  id: number;
  checkId: number;
  title: string;
  language: LanguageDTO;
  numberOfPossibleOutcomesToShow: number;
  rangeQuestions: RangeQuestionDTO[];
  imageQuestions: ImageQuestionDTO[];
  possibleOutcomes: PossibleOutcomeDTO[];
  possibleOutcomesDisplayType: PossibleOutcomesDisplayType;
}
